rfcs/0018-glides-trainsheet-events.md

```diff
@@ -0,0 +1,666 @@
+
+In practice, we limit each location to have only one editor, and limit each editor to only one location, but that rule may or may not change and this data in this event doesn't guarantee that.
+
+Note that the author may or may not match the editors in `changes`, since inspectors can change whether other people are editing. For example, when an inspector takes over editing at a location, they may sign out the previous editor. Then they would issue a `stop` change for the previous editor and a `start` change for themself.
+
+Event type: `com.mbta.ctd.glides.editors-changed.v1`
+Fields in the event:
+- `author` (Author): the inspector who made the changes
+- `changes` (array of EditorChange): a list of start and stop editing events
+
+#### EditorChange
+Fields in the object:
+- `type` (string `"start"`|`"stop"`): Whether the editor started or stopped editing.
+- `location` (Location): the location the editor started or stopped managing.
+- `emailAddress` (string): the e-mail of the editor.
+- `badgeNumber` (string, optional): the badge number of the editor.
+
+
+### Operator signed in
+At the start of their shift, operators need to confirm that they are fit-for-duty and do not have any electronic devices. They currently do this by physically signing a paper trainsheet: in the future, they will do this digitally.
+
+Event type: `com.mbta.ctd.glides.operator_signed_in.v1`
+Fields in the event:
+- `author` (Author): the inspector who signed in the operator
+- `operator` (string): the badge number of the operator who signed in
+- `signedInAt` (RFC3999 timestamp): the time at which they signed in (separate from the `time` of the event)
+- `confirmation` (string): how the operator confirmed that they signed in. Current possibilities are `"type:<badge number>"` if they signed by typing in their badge number, or `"tap"` if they signed in by tapping their badge on an RFID reader, but future string formats may be added without warning, and consumers SHOULD NOT depend on any specific format.
+
+The badge's RFID serial number is not included in the stream for security reasons.
+
+### Trips Updated
+Glides has new information about trips. Consumers who want to know what trains are running should pay attention to this event, and don't need to pay attention to any other events.
+
+Event type: `com.mbta.ctd.glides.trips_updated.v1`
+Fields in the event:
+- `author` (Author): the inspector who inputed the new information.
+- `inputType` (string): the action the inspector performed in Glides that caused the update. There are many ways to update trips in Glides, and all updated information normalized into the same format in `tripUpdates`. This field indicates what the inspector was doing in Glides, for consumers who care not just about service, but also care about how inspectors are using Glides. Example values are `"add-trip"` or `"manage-headways"`, but no guarantees are given about what strings will be used.
```

Comment:
I think that's different, since it's not something that the inspector is doing. We don't otherwise indicate changes to the schedule (such as by noting the schedule import) so I think this is okay to leave out. But I'm also fine with making `author` optional in the case where the update is happening due to an automatic process.

Reply:
It was pretty clean to make `author` optional in the new `metadata` object. All the `metadata` fields are now optional with very weak guarantees.

`inputType` is still named as is, but if a non-human event type wouldn't fit there, it could be in a new optional field in `metadata`.